src/Test/ResetDatabase.php

```diff
@@ -18,23 +21,22 @@ trait ResetDatabase
      */
     public static function _resetDatabase(): void
     {
-        if (DatabaseResetter::hasBeenReset()) {
-            return;
-        }
-
         if (!\is_subclass_of(static::class, KernelTestCase::class)) {
             throw new \RuntimeException(\sprintf('The "%s" trait can only be used on TestCases that extend "%s".', __TRAIT__, KernelTestCase::class));
         }
 
         if ($isDAMADoctrineTestBundleEnabled = DatabaseResetter::isDAMADoctrineTestBundleEnabled()) {
             // disable static connections for this operation
+            // :warning: the kernel should not be booted before calling this!
```

Comment:
I've literally spent hours on this and I figured out that the kernel should not be booted before calling `StaticDriver::setKeepStaticConnections(false);`. 

Since the `shouldReset()` must access to the config, we need the kernel to be booted before calling it.

At first I was trying to do something like this.
```php
        $kernel = static::createKernel();
        $kernel->boot();

        if (!self::shouldReset($kernel)) {
            // $kernel->shutdown(); // this does not work
            return;
        }

        if ($isDAMADoctrineTestBundleEnabled = DatabaseResetter::isDAMADoctrineTestBundleEnabled()) {
            // disable static connections for this operation
            StaticDriver::setKeepStaticConnections(false);

           // $kernel->boot(); // this does not work
        }

        DatabaseResetter::resetDatabase($kernel);
```
and all my tests resulted in errors such as:
> PDOException: SQLSTATE[42000]: Syntax error or access violation: 1305 SAVEPOINT DOCTRINE2_SAVEPOINT_4 does not exist